front/components/assistant/conversation/input_bar/InputBar.tsx

```diff
@@ -0,0 +1,328 @@
+import { Button, Citation, StopIcon } from "@dust-tt/sparkle";
+import { WorkspaceType } from "@dust-tt/types";
+import { AgentConfigurationType } from "@dust-tt/types";
+import { AgentMention, MentionType } from "@dust-tt/types";
+import {
+  createContext,
+  Fragment,
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import { mutate } from "swr";
+
+import { GenerationContext } from "@app/components/assistant/conversation/GenerationContextProvider";
+import InputBarContainer, {
+  InputBarContainerProps,
+} from "@app/components/assistant/conversation/input_bar/InputBarContainer";
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { compareAgentsForSort } from "@app/lib/assistant";
+import { handleFileUploadToText } from "@app/lib/client/handle_file_upload";
+import { useAgentConfigurations } from "@app/lib/swr";
+import { classNames } from "@app/lib/utils";
+
+// AGENT MENTION
+
+function AgentMention({
+  agentConfiguration,
+}: {
+  agentConfiguration: AgentConfigurationType;
+}) {
+  return (
+    <div
+      className={classNames("inline-block font-medium text-brand")}
+      contentEditable={false}
+      data-agent-configuration-id={agentConfiguration?.sId}
+      data-agent-name={agentConfiguration?.name}
+    >
+      @{agentConfiguration.name}
+    </div>
+  );
+}
+
+// Custom hook to synchronize state with a ref as a temporary workaround.
+// This hack is used to maintain a synchronous and up-to-date reference to state values
+// that are otherwise affected by parent component re-renders due to changing dependencies.
+// This approach ensures that the latest state is accessible synchronously when needed
+// for operations that cannot rely on the normal asynchronous nature of setState.
+// This is not an ideal solution and should be addressed with a more robust state management strategy
+// as a follow-up action to minimize unnecessary re-renders and improve component performance.
+function useSyncedState<T>(initialValue: T | undefined) {
```

Comment:
It's far from ideal, but this requires some more refactoring since the deps are not stable which leads to re-rendering. Also from the react documentation:
> The set function only updates the state variable for the next render. If you read the state variable after calling the set function, [you will still get the old value](https://react.dev/reference/react/useState#ive-updated-the-state-but-logging-gives-me-the-old-value) that was on the screen before your call.